### DISCUSSION TITLE
Are here any ENV for define between build time / SSR / CSR

### DISCUSSION THREAD

**Discussion by Cow258** [vercel/next.js#44742] [Help] [answered]
I'm working on something it only run on build time and not run on SSR or CSR.
Any idea? 👀
thanks a lot.

**Comment by icyJoseph** [comment]
Depends on what you want to do. 

There's something called phases, which could be used, but I'd need more information about what you want to accomplish.

**Comment by Cow258** [comment, marked as answer]
```js
export const isBrowser = () => typeof window !== 'undefined'
export const isServer = () => typeof window === 'undefined'
export const isBuildTime = () => process.env.NEXT_PHASE === 'phase-production-build'
```

**Reply by GabenGar** [reply]
Do you really need them as functions? None of these values can change during the runtime so it's better to store them directly even if for better tree-shaking.

**Reply by Cow258** [reply]
You right.
```js
export const isBrowser = typeof window !== 'undefined'
export const isServer = typeof window === 'undefined'
export const isBuildTime = process.env.NEXT_PHASE === 'phase-production-build'
```
